pzsd_bot/cogs/points.py

```diff
@@ -200,18 +222,32 @@ async def fetch_leaderboard(
                 .where(*args)
                 .group_by(pzsd_user.c.id)
             )
-            leaderboard = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+            sorted_points = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+
+        logger.info("Leaderboard length is %s", len(sorted_points))
+
+        leaderboard = [
+            (rank, name, points) for rank, (name, points) in enumerate(sorted_points, 1)
+        ]
+
+        if paginate:
+            iterator = iter(leaderboard)
```

Comment:
And then you don't have to do this if you declare as a generator from the get-go